identity.go

```diff
@@ -110,6 +110,45 @@ func (i *IdentityIterator) Next() bool {
 	return true
 }
 
+// WriteTo encodes and writes all IdentityInfos to w.
+// It returns the number of bytes written to w and
+// the first error encounterred, if any.
+func (i *IdentityIterator) WriteTo(w io.Writer) (int64, error) {
+	type Response struct {
+		Identity  Identity  `json:"identity"`
+		Policy    string    `json:"policy,omitempty"`
+		CreatedAt time.Time `json:"created_at,omitempty"`
+		CreatedBy Identity  `json:"created_by,omitempty"`
+
+		Err string `json:"error,omitempty"`
+	}
+	if i.err != nil || i.closed {
+		return 0, i.err
+	}
+
+	cw := countWriter{W: w}
+	encoder := json.NewEncoder(&cw)
+	for {
+		var resp Response
+		if err := i.decoder.Decode(&resp); err != nil {
```

Comment:
So this is stateful? Meaning it will only return whatever hasn't been iterated over already? That would be a nice thing to have in the function docs, since that may be unexpected to some.

Reply:
Maybe just:
```
// WriteTo encodes and writes all remaining IdentityInfos from the current iterator position to w.
```